src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/DeductionReasonCode.java

```diff
@@ -0,0 +1,44 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+import org.apache.commons.lang3.EnumUtils;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.EdifactValidationException;
+
+import java.util.Objects;
+
+@EqualsAndHashCode(callSuper = false)
+@Builder
+@Data
```

Comment:
Make `code` final and only `@RequiredArgsConstructor` will be needed

Reply:
done